### PR TITLE
[TorchOnnxToTorch] Improve GroupQueryAttention lowering

### PR DESCRIPTION
**Pull request by IanWood1**
Refactor GroupQueryAttention conversion to use reshape+transpose instead of reshape-only for Q/K/V head separation. Add support for packed QKV input, fix seqlens_k causal mask boundary calculation, improve position ID generation for rotary embeddings, and properly order SDPA after KV cache concatenation. Add tests for past KV, mask boundaries, multi-token prefill, and rotary position IDs.

### CODE REVIEW

File: lib/Conversion/TorchOnnxToTorch/ComMicrosoftDomain.cpp [hunk 1]
```diff
@@ -389,71 +487,247 @@ void mlir::torch::onnx_c::populateComMicrosoftDomain(
...
 
-        // Do attention.
+        // Compute present_key and present_value by concatenating past with
+        // current. These are used both for attention AND as output for the next
+        // iteration's KV cache.
+        // present_key = torch.cat([past_key, key], dim=2)
+        // present_value = torch.cat([past_value, value], dim=2)
+        // Always concatenate, even if past_key/past_value are empty (sequence
+        // length 0), as concatenating empty tensors still produces correct
+        // results.
```
Comment (IanWood1) [inline lib/Conversion/TorchOnnxToTorch/ComMicrosoftDomain.cpp:503]: This needs to be worked out before landing
